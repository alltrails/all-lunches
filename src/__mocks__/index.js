export { default as restaurantMock } from './restaurant.mock';
export { default as restaurantsMock } from './restaurants.mock';
export { default as reactRefMock } from './reactRef.mock';
