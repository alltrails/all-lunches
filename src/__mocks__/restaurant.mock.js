/* eslint-disable sort-keys  */

export default {
  coordinates: [-122.408999, 37.7902566],
  id: 'ChIJkVLoONZ_j4ARx2vJ4umERnI',
  name: "Uncle Vito's Pizza",
  photoUrl:
    'https://maps.googleapis.com/maps/api/place/js/PhotoService.GetPhoto?1sAap_uEBdIFbWc8LjoO8hOGn1QTJBeq21XOCk6ZHFGBmGWWqrImCdbERWb-K4T9Mq_QB5Mspi1Kiyc_GTU_XRbcReDLyGaIWdJWcssnsj3mZmy5xQsGCPeuOiBjjDEciT92r5iUeEAvSNn4dPN2r9rng9SH3fTXpxlT6BA5TB1MPzYfQ_GU5J&3u4032&5m1&2e1&callback=none&key=AIzaSyDg0Zd8r5Jf5i4MWlM9_eGYrpyRUeqBbPY&token=10623',
  placeUrl:
    "https://www.google.com/maps/search/?api=1&query=Uncle%20Vito's%20Pizza&query_place_id=ChIJkVLoONZ_j4ARx2vJ4umERnI",
  priceLevel: 1,
  rating: 4.3,
  supportingText: 'restaurant',
  userRatingsTotal: 1427,
};
