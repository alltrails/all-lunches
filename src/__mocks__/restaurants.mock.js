import DefaultStorePNG from 'assets/store-front.png';

export default [
  {
    coordinates: [-122.408999, 37.7902566],
    id: 'ChIJkVLoONZ_j4ARx2vJ4umERnI',
    name: "Uncle Vito's Pizza",
    photoUrl: DefaultStorePNG,
    placeUrl:
      "https://www.google.com/maps/search/?api=1&query=Uncle%20Vito's%20Pizza&query_place_id=ChIJkVLoONZ_j4ARx2vJ4umERnI",
    priceLevel: 1,
    rating: 4.3,
    supportingText: 'restaurant',
    userRatingsTotal: 1427,
  },
  {
    coordinates: [-122.408675, 37.7900735],
    id: 'ChIJg7cNIYyAhYAR8D4ml1oItTg',
    name: "Roxanne's Cafe",
    photoUrl: DefaultStorePNG,
    placeUrl:
      "https://www.google.com/maps/search/?api=1&query=Roxanne's%20Cafe&query_place_id=ChIJg7cNIYyAhYAR8D4ml1oItTg",
    priceLevel: 2,
    rating: 4.2,
    supportingText: 'restaurant',
    userRatingsTotal: 1219,
  },
  {
    coordinates: [-122.4073542, 37.7902509],
    id: 'ChIJaa9OAYyAhYARxwgUHLNEpOY',
    name: 'Bouche',
    photoUrl: DefaultStorePNG,
    placeUrl:
      'https://www.google.com/maps/search/?api=1&query=Bouche&query_place_id=ChIJaa9OAYyAhYARxwgUHLNEpOY',
    priceLevel: 3,
    rating: 4.5,
    supportingText: 'bar',
    userRatingsTotal: 465,
  },
  {
    coordinates: [-122.409088, 37.7902449],
    id: 'ChIJvXXIOoyAhYARe-ixq3WdmrY',
    name: 'Sons & Daughters',
    photoUrl: DefaultStorePNG,
    placeUrl:
      'https://www.google.com/maps/search/?api=1&query=Sons%20%26%20Daughters&query_place_id=ChIJvXXIOoyAhYARe-ixq3WdmrY',
    priceLevel: 4,
    rating: 4.6,
    supportingText: 'restaurant',
    userRatingsTotal: 249,
  },
];
