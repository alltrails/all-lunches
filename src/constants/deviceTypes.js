export const DESKTOP = 'desktop';
export const MOBILE = 'mobile';
export const TABLET = 'tablet';
