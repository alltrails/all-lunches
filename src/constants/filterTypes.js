export const HIGH_TO_LOW_FILTER_OPTION = 'high-to-low';
export const LOW_TO_HIGH_FILTER_OPTION = 'low-to-high';
