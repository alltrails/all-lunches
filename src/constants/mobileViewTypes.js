export const MAP_VIEW = 'map_view';
export const LIST_VIEW = 'list_view';
