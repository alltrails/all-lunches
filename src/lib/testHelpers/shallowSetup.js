/* eslint-disable react/jsx-props-no-spreading */
import React from 'react';
import { shallow } from 'enzyme';

export default (Component, defaultProps) =>
  (otherProps = {}) => {
    const props = { ...defaultProps, ...otherProps };
    const wrapper = shallow(<Component {...props} />);

    return { props, wrapper };
  };
