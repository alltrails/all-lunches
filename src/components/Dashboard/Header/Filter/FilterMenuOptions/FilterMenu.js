import React from 'react';
import PropTypes from 'prop-types';

import { refType } from 'constants/propTypes';
import * as FilterTypes from 'constants/filterTypes';

import CheckmarkIcon from 'components/svg/CheckmarkIcon';
import FocusTrap from 'components/shared/FocusTrap';
import {
  ApplyButton,
  CheckmarkIconWrapper,
  Form,
  ListWrapper,
  OptionItem,
  OptionItemButton,
  Options,
} from './style';

const menuOptions = [
  { id: FilterTypes.HIGH_TO_LOW_FILTER_OPTION, title: 'Ratings High to Low' },
  { id: FilterTypes.LOW_TO_HIGH_FILTER_OPTION, title: 'Ratings Low to High' },
];

const FilterMenu = ({ menuRef, onMenuOptionClick, onSubmit, selectedMenuOption }) => (
  <FocusTrap fallbackFocus="#pop-out-menu">
    <Options ref={menuRef} tabIndex="-1" id="pop-out-menu">
      <Form>
        <ListWrapper>
          {menuOptions.map((option) => {
            const isSelected = selectedMenuOption === option.id;

            return (
              <OptionItemButton
                key={option.id}
                onClick={() => onMenuOptionClick(option.id)}
                type="button"
              >
                <OptionItem>
                  <CheckmarkIconWrapper isSelected={isSelected}>
                    <CheckmarkIcon
                      titleText={option.title}
                      variant={isSelected ? 'green' : 'white'}
                    />
                  </CheckmarkIconWrapper>
                  {option.title}
                </OptionItem>
              </OptionItemButton>
            );
          })}
        </ListWrapper>
        <ApplyButton disabled={!selectedMenuOption} onClick={onSubmit}>
          Apply
        </ApplyButton>
      </Form>
    </Options>
  </FocusTrap>
);
FilterMenu.propTypes = {
  menuRef: refType,
  onMenuOptionClick: PropTypes.func.isRequired,
  onSubmit: PropTypes.func.isRequired,
  selectedMenuOption: PropTypes.string,
};

FilterMenu.defaultProps = {
  menuRef: null,
  selectedMenuOption: null,
};

export default FilterMenu;
