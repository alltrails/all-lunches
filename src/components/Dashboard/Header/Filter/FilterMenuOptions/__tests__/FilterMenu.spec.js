/* eslint-env jest */

import shallowSetup from 'lib/testHelpers/shallowSetup';
import * as FilterTypes from 'constants/filterTypes';
import { reactRefMock } from '__mocks__';

import FilterMenu from '../FilterMenu';

describe('FilterMenu', () => {
  const defaultProps = {
    menuRef: reactRefMock,
    onMenuOptionClick: () => {},
    onSubmit: () => {},
    selectedMenuOption: null,
  };
  const setup = shallowSetup(FilterMenu, defaultProps);

  it('renders the expected view', () => {
    const { wrapper } = setup();
    expect(wrapper).toMatchSnapshot();
  });

  it('renders the expected view when menu option is selected', () => {
    const { wrapper } = setup({ selectedMenuOption: FilterTypes.LOW_TO_HIGH_FILTER_OPTION });
    expect(wrapper).toMatchSnapshot();
  });
});
