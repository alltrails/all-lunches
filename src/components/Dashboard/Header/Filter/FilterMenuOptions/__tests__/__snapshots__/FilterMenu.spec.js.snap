// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterMenu renders the expected view 1`] = `
<FocusTrap
  clickOutsideDeactivates={true}
  fallbackFocus="#pop-out-menu"
>
  <style__Options
    id="pop-out-menu"
    tabIndex="-1"
  >
    <style__Form>
      <style__ListWrapper>
        <style__OptionItemButton
          key="high-to-low"
          onClick={[Function]}
          type="button"
        >
          <style__OptionItem>
            <style__CheckmarkIconWrapper
              isSelected={false}
            >
              <CheckmarkIcon
                titleText="Ratings High to Low"
                variant="white"
              />
            </style__CheckmarkIconWrapper>
            Ratings High to Low
          </style__OptionItem>
        </style__OptionItemButton>
        <style__OptionItemButton
          key="low-to-high"
          onClick={[Function]}
          type="button"
        >
          <style__OptionItem>
            <style__CheckmarkIconWrapper
              isSelected={false}
            >
              <CheckmarkIcon
                titleText="Ratings Low to High"
                variant="white"
              />
            </style__CheckmarkIconWrapper>
            Ratings Low to High
          </style__OptionItem>
        </style__OptionItemButton>
      </style__ListWrapper>
      <style__ApplyButton
        disabled={true}
        onClick={[Function]}
      >
        Apply
      </style__ApplyButton>
    </style__Form>
  </style__Options>
</FocusTrap>
`;

exports[`FilterMenu renders the expected view when menu option is selected 1`] = `
<FocusTrap
  clickOutsideDeactivates={true}
  fallbackFocus="#pop-out-menu"
>
  <style__Options
    id="pop-out-menu"
    tabIndex="-1"
  >
    <style__Form>
      <style__ListWrapper>
        <style__OptionItemButton
          key="high-to-low"
          onClick={[Function]}
          type="button"
        >
          <style__OptionItem>
            <style__CheckmarkIconWrapper
              isSelected={false}
            >
              <CheckmarkIcon
                titleText="Ratings High to Low"
                variant="white"
              />
            </style__CheckmarkIconWrapper>
            Ratings High to Low
          </style__OptionItem>
        </style__OptionItemButton>
        <style__OptionItemButton
          key="low-to-high"
          onClick={[Function]}
          type="button"
        >
          <style__OptionItem>
            <style__CheckmarkIconWrapper
              isSelected={true}
            >
              <CheckmarkIcon
                titleText="Ratings Low to High"
                variant="green"
              />
            </style__CheckmarkIconWrapper>
            Ratings Low to High
          </style__OptionItem>
        </style__OptionItemButton>
      </style__ListWrapper>
      <style__ApplyButton
        disabled={false}
        onClick={[Function]}
      >
        Apply
      </style__ApplyButton>
    </style__Form>
  </style__Options>
</FocusTrap>
`;
